tests/components/devialet/test_media_player.py

```diff
@@ -246,57 +272,170 @@ async def test_media_player_offline(
     assert entry.state is ConfigEntryState.NOT_LOADED
 
 
+async def test_device_shutdown(
+    hass: HomeAssistant, aioclient_mock: AiohttpClientMocker
+) -> None:
+    """Test the Devialet configuration entry loading and the device going offline."""
+    with patch.object(DevialetApi, "upnp_available", return_value=True):
+        entry = await setup_integration(hass, aioclient_mock, serial=None)
+
+        assert entry.state is ConfigEntryState.LOADED
+        assert entry.unique_id is None
+
+        await hass.config_entries.async_unload(entry.entry_id)
+        await hass.async_block_till_done()
+
+        assert entry.state is ConfigEntryState.NOT_LOADED
+
+
 async def test_media_player_without_serial(
     hass: HomeAssistant, aioclient_mock: AiohttpClientMocker
 ) -> None:
     """Test the Devialet configuration entry loading and unloading."""
-    entry = await setup_integration(hass, aioclient_mock, serial=None)
+    with patch.object(DevialetApi, "upnp_available", return_value=True):
+        entry = await setup_integration(hass, aioclient_mock, serial=None)
 
-    assert entry.state is ConfigEntryState.LOADED
-    assert entry.unique_id is None
+        assert entry.state is ConfigEntryState.LOADED
+        assert entry.unique_id is None
 
-    await hass.config_entries.async_unload(entry.entry_id)
-    await hass.async_block_till_done()
+        await hass.config_entries.async_unload(entry.entry_id)
+        await hass.async_block_till_done()
 
-    assert entry.state is ConfigEntryState.NOT_LOADED
+        assert entry.state is ConfigEntryState.NOT_LOADED
 
 
-async def test_media_player_services(
-    hass: HomeAssistant, aioclient_mock: AiohttpClientMocker
+async def test_browse_media(
+    hass: HomeAssistant,
+    hass_ws_client: WebSocketGenerator,
+    aioclient_mock: AiohttpClientMocker,
 ) -> None:
     """Test the Devialet services."""
+
     entry = await setup_integration(
         hass, aioclient_mock, state=MediaPlayerState.PLAYING
     )
-
     assert entry.state is ConfigEntryState.LOADED
 
-    target = {ATTR_ENTITY_ID: hass.states.get(f"{MP_DOMAIN}.{NAME}").entity_id}
+    client = await hass_ws_client(hass)
+
+    with (
+        patch.object(DevialetApi, "upnp_available", return_value=True),
+        patch(
+            "homeassistant.components.media_source.async_browse_media",
+            return_value=True,
+        ) as mock_browse_media,
```

Comment:
Why do we patch this internal?

Reply:
What would be another solution to test the async browse media function?

Reply:
I don't know, I did work on the Spotify one, but I am more wondering why we do this so I can see how we would solve this otherwise, hence its just a question rather than a "this should be changed" :)

Reply:
I tried to patch the first function after the Devialet integration's call. So that's why I've patched this one. Not sure if it's the best way to test it but it seems to work.